### ISSUE TITLE
Public IPs of Azure virtual machines in scale sets are not exposed by the `__meta_azure_machine_public_ip` label in `azure_sd_config`

### ISSUE THREAD

**Issue by lisushka** [prometheus/prometheus#9293]
**What did you do?**

We have Azure virtual machines that are part of a virtual machine scale set.  We attempted to scrape metrics from them using the following `azure_sd_configs` block:
```
azure_sd_configs:
  - environment: AzurePublicCloud
    port: 9182
    subscription_id: ***
    tenant_id: ***
    client_id: ***
    client_secret: ***
    refresh_interval: 5m
    authentication_method: OAuth
  relabel_configs:
  - source_labels: [__meta_azure_machine_tag_prometheus]
    separator: ;
    regex: true.*
    replacement: $1
    action: keep
  - source_labels: [__meta_azure_machine_tag_node_exporter]
    separator: ;
    regex: true.*
    replacement: $1
    action: keep
  - source_labels: [__meta_azure_machine_name]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
```

**What did you expect to see?**

We expected that the public IP addresses of the virtual machines would be visible in Prometheus under the `__meta_azure_machine_public_ip` label.

**What did you see instead? Under which circumstances?**

The private IP addresses of the virtual machines are visible under the `__meta_azure_machine_private_ip` tag, but the public IP addresses are not visible:

<img width="1003" alt="Screenshot of Azure console showing partially consored virtual machine public and private IP addresses" src="https://user-images.githubusercontent.com/28942094/131953807-4578adfe-9fe2-4b95-bcde-21adf8a5d9a7.png">
<img width="625" alt="Screenshot of Prometheus labels for the same image, showing the private IP address label but not the public IP address label" src="https://user-images.githubusercontent.com/28942094/131953812-a56f5665-1c5b-4be4-a0b4-21e95e08f67b.png">

* Prometheus version:

	We have tested with both `prometheus:v2.29.2` and `prometheus:v2.20.1`.

* Prometheus configuration file:
```
azure_sd_configs:
  - environment: AzurePublicCloud
    port: 9182
    subscription_id: ***
    tenant_id: ***
    client_id: ***
    client_secret: ***
    refresh_interval: 5m
    authentication_method: OAuth
  relabel_configs:
  - source_labels: [__meta_azure_machine_tag_prometheus]
    separator: ;
    regex: true.*
    replacement: $1
    action: keep
  - source_labels: [__meta_azure_machine_tag_node_exporter]
    separator: ;
    regex: true.*
    replacement: $1
    action: keep
  - source_labels: [__meta_azure_machine_name]
    separator: ;
    regex: (.*)
    target_label: instance
    replacement: $1
    action: replace
```

**Comment by roidelapluie** [comment]
This sounds like a bug, do you have API output to help us debugging this by any chance? Thanks.

**Comment by gavinkflam** [comment]
Hi, I worked on a fix and created a PR #13241. Please help to review the changes.

**Comment by bboreham** [comment]
Closed by #13241.
